src/pages/iv-opening-cost/Forms/OpeningCostForm.js

```diff
@@ -0,0 +1,163 @@
+import { ResourceLookup } from 'src/components/Shared/ResourceLookup'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+
+export default function OpeningCostForm({ labels, maxAccess, record }) {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const recordId = String(record?.year) + String(record?.itemId) || null;
+
+  const invalidate = useInvalidate({
+    endpointId: InventoryRepository.OpeningCost.qry
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: null,
+      itemId: null,
+      year: '',
+      avgCost: 0
+    },
+    maxAccess,
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      year: yup.number().required(),
+      itemId: yup.string().required(),
+      avgCost: yup.number().min(0).max(999999999)
+    }),
+    onSubmit: async obj => {
+      await postRequest({
+        extension: InventoryRepository.OpeningCost.set,
+        record: JSON.stringify(obj)
+      })
+
+      if (!obj.recordId) {
+        toast.success(platformLabels.Added)
+
+        formik.setValues({
+          ...obj,
+          recordId: String(obj.year) + String(obj.itemId)
+        })
+      } else toast.success(platformLabels.Edited)
+
+      invalidate()
+    }
+  })
+
+  const editMode = !!formik.values.recordId
+
+  useEffect(() => {
+    ;(async function () {
+      if (record?.itemId && record?.year && recordId) {
+        const res = await getRequest({
+          extension: InventoryRepository.OpeningCost.get,
+          parameters: `_itemId=${record?.itemId}&_fiscalYear=${record?.year}`
+        })
+
+        formik.setValues({
+          ...res.record,
+          recordId: String(res.record.year) + String(res.record.itemId)
+        })
+      }
+    })()
+  }, [])
+
+  return (
+    <FormShell 
+      resourceId={ResourceIds.InventoryOpeningCosts} 
+      form={formik} 
+      maxAccess={maxAccess} 
+      editMode={editMode}
+      infoVisible={false}
+      isSavedClear={false}
+      isInfo={false}
+      isCleared={false}
+    >
+      <VertLayout>
+        <Grow>
+          <Grid container spacing={4}>
+            <Grid item xs={12}>
+              <ResourceComboBox
+                endpointId={SystemRepository.FiscalYears.qry}
+                readOnly={editMode}
+                name='year'
+                label={labels?.fiscalYear}
+                valueField='fiscalYear'
+                displayField='fiscalYear'
+                values={formik.values}
+                required
+                maxAccess={maxAccess}
+                onChange={(event, newValue) => {
+                  formik.setFieldValue('year', newValue?.fiscalYear)
+                }}
```

Comment:
you need to add onClear function or in the sublit use || '' because when you choose a value then remove it it acts like you didn't clear the field
![image](https://github.com/user-attachments/assets/0642f4c1-ab89-4475-b712-e6cb9463fac9)